domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt

```diff
@@ -231,11 +231,79 @@ class TopicListControllerTest {
   }
 
   @Test
-  fun testGetPromotedStoryList_markChapDoneFracStory0Exp0_ongoingStoryListIsCorrect() {
+  fun testGetPromotedActivityList_markFracStory0Exp0InProgressSaved_ongoingStoryListIsCrt() {
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId0,
+      timestampOlderThanOneWeek = false
+    )
+
+    val promotedActivityList = retrievePromotedActivityList()
+    assertThat(promotedActivityList.promotedStoryList.recentlyPlayedStoryCount)
+      .isEqualTo(1)
+    verifyOngoingStoryAsFractionStory0Exploration0(
+      promotedActivityList.promotedStoryList.recentlyPlayedStoryList[0]
+    )
+  }
+
+  @Test
+  fun testGetPromotedActivityList_markRecentlyPlayedFracStory0Exp0_ongoingStoryListIsCrt() {
+    storyProgressTestHelper.markInProgressNotSavedFractionsStory0Exp0(
+      profileId0,
+      timestampOlderThanOneWeek = false
+    )
+
+    val promotedActivityList = retrievePromotedActivityList()
+    assertThat(promotedActivityList.promotedStoryList.recentlyPlayedStoryCount)
+      .isEqualTo(1)
+    verifyOngoingStoryAsFractionStory0Exploration0(
+      promotedActivityList.promotedStoryList.recentlyPlayedStoryList[0]
+    )
+  }
+
+  @Test
+  fun testGetPromotedStoryList_markChapDoneFracStory0Exp0_ongoingStoryListIsCrt() {
```

Comment:
This part slightly confuses me, we used to call those chapters as recently played which were marked as `STARTED_NOT_COMPLETED` but now a lesson that was left partially complete by the learner can be marked as either 
- `STARTED_NOT_COMPLETED` (if the lesson was left partially completed before we introduced checkpointing)
- `IN_PROGRESS_SAVED`
- `IN_PROGRESS_NOT_SAVED`

So I think we should call a lesson/story/topic as recently played if the lesson has been left partially complete by the user. ( has one or more partially complete lessons in case of topic/story)

Reply:
Following this above explanation, this test checks that since we have only one lesson of the story 0 of fractions topic partially complete, so the size of onGoingStoryList should be equal to 1.

Reply:
Ah, so the main issue here (& the reason we need to keep all the "started_not_completed" stuff) is for interoperability? If that's the case, I actually lean toward us just dropping the enum entirely and, when reading progress, revert the state back to NOT_COMPLETED if we encounter that type. This reduces a pretty large surface area that we need to maintain in the codebase, and given that we don't have very many users of the app it should have small impact on the existing userbase. It also self-corrects reasonably well when you start playing lessons again.

Longer-term (e.g. the pathway from lightweight to full checkpointing) we will need to ensure interoperability, but given the cost here it doesn't seem worth the complexity given the app is still pre-release.

Reply:
@BenHenning can we change the lessons that are STARTED_NOT_COMPLETED to IN_PROGRESS_NOT_SAVED because these lessons were started but do not have any progress saved for them.

Reply:
Dropping the enum would just mean the following, 
- removing the function that marks lessons as started not completed
- removing started not completed lessons from when creating the promoted story list
- removing all the tests that check app behaviour when lesson is marked as started_not_completed.

Apart from the above points I think there should not be any other work (since this PR already has those tests for in_progress_saved/not_saved which we will be removing for started_not_completed.) 

So I think we should drop started_not_completed in a separate PR while fixing this issue #3479  because lessons will be marked as in_progress_saved/not_saved only after that is merged (until then we will be marking lessons as started_not_completed)

Reply:
> @BenHenning Do you mean we change the state of the lessons marked as STARTED_NOT_COMPLETED to NOT_STARTED? if so can we change the lessons that are STARTED_NOT_COMPLETED to IN_PROGRESS_NOT_SAVED because these lessons were started but do not have any progress saved for them?

Yes, this is exactly what I meant.



> Dropping the enum would just mean the following,
> 
> * removing the function that marks lessons as started not completed in storyProgressController
> * removing started not completed lessons from the promoted story list
> * removing all the tests that check app behaviour when the lesson is marked as started_not_completed.
> 
> Apart from the above points, I think there should not be any other work (since this PR already has those tests for in_progress_saved/not_saved which we will be removing for started_not_completed.)
> 
> So I think we should drop started_not_completed in a separate PR while fixing this issue #3479 because lessons will be marked as in_progress_saved/not_saved only after that is merged (until then we will be marking lessons as started_not_completed).
> 
> @BenHenning WDYT?

Fixing this as part of #3479 sounds good to me, so long as it does get fixed. Could you file an issue to explicitly track this particular task & make sure it gets tracked in your project accordingly? It can be resolved in the same PR that resolves #3479, but being explicit with a separate task will help ensure we can track the solution.

Reply:
Filed a new issue (#3662) to solve this. Also added todos.